bridges/bin/millau/runtime/src/lib.rs

```diff
@@ -531,6 +531,7 @@ pub type SignedBlock = generic::SignedBlock<Block>;
 pub type BlockId = generic::BlockId<Block>;
 /// The `SignedExtension` to the basic transaction logic.
 pub type SignedExtra = (
+	frame_system::CheckNonZeroSender<Runtime>,
```

Comment:
why is this added the top? is it a requirement?

Reply:
IIRC `SignedExtension`s get run in the order of declaration, so we'd definitely want to check for the non-zero sender somewhere early. What's the problem that you would run into if it's at the top?

Reply:
As you can see in this PR, it changes the index of nonce and tip. Not a major issue, but still some additional changes requires more work.

Will it make any visible difference if it is placed after check weight before charge transaction fee?

Reply:
This shouldn't affect the transaction format if that is what your are implying. This check should not affect the signature payload since the extension itself does not contain any data.

Reply:
No. I mean it will require us to change code like https://github.com/AcalaNetwork/Acala/blob/a657f2db621a7d2364c168da666b36d0d1ee0d84/runtime/mandala/src/lib.rs#L1975

Reply:
We should probably switch to using a struct with named fields rather than a tuple here. This will likely keep happening.